packages/schema-type/src/typescript/arrayAbiTypes.ts

```diff
@@ -1,17 +1,42 @@
 import { StaticAbiType } from "./schemaAbiTypes";
+import { isStaticAbiType } from "./staticAbiTypes";
 
-const arrayAbiTypePattern = /\[\]$/;
+export type ArrayAbiType = `${StaticAbiType}[]`;
+export type FixedArrayAbiType = `${StaticAbiType}[${number}]`;
+
+const arrayPattern = /\[\]$/;
+const fixedArrayPattern = /\[\d+\]$/;
 
 export type ArrayAbiTypeToStaticAbiType<T extends string> = T extends `${infer StaticAbiType}[]`
   ? StaticAbiType
   : never;
 
-export function isArrayAbiType<T extends `${StaticAbiType}[]`>(abiType: string): abiType is T {
-  return arrayAbiTypePattern.test(abiType);
+export type FixedArrayAbiTypeToStaticAbiType<T extends string> = T extends `${infer StaticAbiType}[${number}]`
+  ? StaticAbiType
+  : never;
+
+export function isArrayAbiType<T extends ArrayAbiType>(abiType: string): abiType is T {
```

Comment:
i would expect a type guard like this to accept an `unknown` input, otherwise typescript will complain if your input might not be an `ArrayAbiType`, which is what the type guard is supposed to check

Reply:
this was already like this but yeah, can expand to match the other `isStaticAbiType`, etc helpers in this package

Reply:
oh I see what you mean now with the comment below